source/Octopus.Tentacle/Scripts/Kubernetes/RunningKubernetesJobScript.cs

```diff
@@ -0,0 +1,323 @@
+                            Containers = new List<V1Container>
+                            {
+                                new()
+                                {
+                                    Name = jobName,
+                                    Image = await GetDefaultContainer(),
+                                    Command = new List<string> { "bash" },
+                                    Args = new List<string>
+                                    {
+                                        $"/data/tentacle-home/{instanceName}/Work/{scriptTicket.TaskId}/bootstrapRunner.sh",
+                                        $"/data/tentacle-home/{instanceName}/Work/{scriptTicket.TaskId}",
+                                        $"/data/tentacle-home/{instanceName}/Work/{scriptTicket.TaskId}/{scriptName}"
+                                    }.Concat((workspace.ScriptArguments ?? Array.Empty<string>())
+                                        .SelectMany(arg => new[]
+                                        {
+                                            $"'{arg}'"
+                                        })).ToList(),
+                                    VolumeMounts = new List<V1VolumeMount>
+                                    {
+                                        new("/data/tentacle-home", "tentacle-home"),
+                                    },
+                                    Env = new List<V1EnvVar>
+                                    {
+                                        new(EnvironmentVariables.TentacleHome, $"/data/tentacle-home/{instanceName}"),
+                                        new(EnvironmentVariables.TentacleVersion, Environment.GetEnvironmentVariable(EnvironmentVariables.TentacleVersion)),
+                                        new(EnvironmentVariables.TentacleCertificateSignatureAlgorithm, Environment.GetEnvironmentVariable(EnvironmentVariables.TentacleCertificateSignatureAlgorithm)),
+                                        new("OCTOPUS_RUNNING_IN_CONTAINER", "Y")
+                                    }
+                                }
+                            },
+                            ServiceAccountName = KubernetesConfig.JobServiceAccountName,
+                            RestartPolicy = "Never",
+                            Volumes = volumes
+                        }
+                    },
+                    BackoffLimit = 0, //we never want to rerun if it fails
+                    TtlSecondsAfterFinished = 1800 //30min
+                }
+            };
+
+            writer.WriteVerbose($"Executing script in Kubernetes Job '{job.Name()}'");
+
+            await jobService.CreateJob(job, cancellationToken);
+        }
+
+        static readonly List<Version> KnownLatestContainerTags = new List<Version>
+        {
+            new(1, 26, 3),
+            new(1, 27, 3),
+            new(1, 28, 2),
+        };
+
+        async Task<string> GetDefaultContainer()
```

Comment:
Minor, consider pulling some parts of this class such as this worker tool container resolution into its own class that can be tested seperately. ALso makes this class easier to grok.

Reply:
Done. Created a new `KubernetesJobContainerResolver` in [a2a48b7](https://github.com/OctopusDeploy/OctopusTentacle/pull/690/commits/a2a48b76b42da74a82b8d99115f1ce3d09c49010)